### PR TITLE
ip_sniffing: Support tshark (wireshark) 3.x

### PR DESCRIPTION
**Pull request by luckyh**
From tshark 3.0.0, the field name - bootp [1] is no longer supported
and users should use dhcp [2] instead.

    tshark: Some fields aren't valid:
        bootp.option.dhcp
        bootp.ip.your
        ...
    (Process terminated with status 1)

Updating the module accordingly.

References:
[1] https://www.wireshark.org/docs/dfref/b/bootp.html
[2] https://www.wireshark.org/docs/dfref/d/dhcp.html

Signed-off-by: Xu Han <xuhan@redhat.com>

ID: 1917682

### CODE REVIEW

**Comment by luckyh** [comment]
@yanan-fu @PaulYuuu request for your review, thanks!

**Comment by luckyh** [comment]
@kylazhang @chunfuwen @chloerh would you please help review the patch as you may be interested as well? thanks!

**Comment by luckyh** [comment]
Test result:

tshark 3.4.0
```
2021-01-20 03:10:31: Running as user "root" and group "root". This could be dangerous.
2021-01-20 03:10:31: ip.src ip.dst dhcp.type dhcp.id dhcp.hw.mac_addr dhcp.ip.your dhcp.option.dhcp ipv6.src ipv6.dst
2021-01-20 03:10:31: Capturing on 'any'
2021-01-20 03:10:34: 0.0.0.0 255.255.255.255 1 0xcd508cbd 9a:55:1b:xx:xx:xx 0.0.0.0 3
2021-01-20 03:10:34: 0.0.0.0 255.255.255.255 1 0xcd508cbd 9a:55:1b:xx:xx:xx 0.0.0.0 3
...
```

tshark 2.6.2
```
2021-01-20 03:27:17: Running as user "root" and group "root". This could be dangerous.
2021-01-20 03:27:17: ip.src ip.dst bootp.type bootp.id bootp.hw.mac_addr bootp.ip.your bootp.option.dhcp ipv6.src ipv6.dst
2021-01-20 03:27:17: Capturing on 'any'
2021-01-20 03:27:21:        fe80::9eb6:54xx:xxxx:xxxx ff02::1:2
2021-01-20 03:27:22:        fe80::fe15:b4xx:xxxx:xxxx ff02::1:2
...
```

**Comment by yanan-fu** [comment]
Test with both tshark 3.4.0 and tshark 2.6.2 env,  can get ip address successfully, and ip-sniffer.log is normally now with this change.  And the change LGTM

Acked-by: Yanan Fu <yfu@redhat.com>

**Review by yanan-fu — APPROVED** [review]

**Comment by chloerh** [comment]
Running an acceptance test job with this pr on both x86 and ppc. 
Will get x86 result within 1 hour. PPC job is queuing for machine.

**Review by chloerh — APPROVED** [review]
All 27 cases on x86 passed.

File: virttest/ip_sniffing.py [hunk 1]
```diff
@@ -323,5 +346,35 @@ def _output_handler(self, line):
             return True
 
 
+class TShark1To2(TSharkSniffer):
+
+    """
+    TShark sniffer class for version 1.0.0 - 2.x.x.
+    """
+
+    options = ("-npi any -T fields -E separator=/s -E occurrence=f "
```
Comment (PaulYuuu) [inline virttest/ip_sniffing.py:355]: I see that except for `bootp` and `dhcp`, other options are the same. Is it better to extract the options and pass `bootp`/`dhcp`?

Comment (luckyh) [reply virttest/ip_sniffing.py:355]: I'm not sure if it is possible, would you like to give an example? thanks.

Comment (PaulYuuu) [reply virttest/ip_sniffing.py:355]: Em, it's possible but need to put it into the `__init__` magic method, which is not worth it. Let's keep it as it is.

**Review by PaulYuuu — APPROVED** [review]

File: virttest/ip_sniffing.py (hunk 1 above)
Comment (luckyh) [reply virttest/ip_sniffing.py:355]: Yep, thanks.

**Comment by yanan-fu** [comment]
@luckyh  @PaulYuuu Do you have any more comment ?   If not, could we merge it since there are two acks already ? Thanks!

**Comment by luckyh** [comment]
> @luckyh @PaulYuuu Do you have any more comment ? If not, could we merge it since there are two acks already ? Thanks!

Sure, thanks all!

**Pull request merged**